DuckDuckGo/RemoteMessagingDebugViewController.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  RemoteMessagingDebugViewController.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+import SwiftUI
+import RemoteMessaging
+import Core
+import Combine
+import Persistence
+
+class RemoteMessagingDebugViewController: UIHostingController<RemoteMessagingDebugRootView> {
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder, rootView: RemoteMessagingDebugRootView())
+    }
+
+}
+
+struct RemoteMessagingDebugRootView: View {
+
+    @ObservedObject var model = RemoteMessagingDebugViewModel()
+
+    var body: some View {
+        List {
+            Section {
+                ForEach(model.messages, id: \.id) { entry in
+                    VStack(alignment: .leading) {
+                        Text(entry.id ?? "")
+                            .font(.system(size: 14))
+                        Text(entry.message ?? "")
+                            .font(.system(size: 12))
+                        Text("Shown: \(String(describing: entry.shown))")
+                            .font(.system(size: 10))
+                        Text("Status: \(statusString(for: entry.status))")
+                            .font(.system(size: 10))
+                    }
+                }
+            } footer: {
+                Text("This list contains messages that have been shown plus at most 1 message that is scheduled for showing. There may be more messages in the config that will be presented, but they haven't been processed yet.")
+            }
+        }
+        .navigationTitle("\(model.messages.count) Remote Messages")
+        .toolbar {
+            if #available(iOS 15, *) {
+                Button("Delete All", role: .destructive) {
+                    model.deleteAll()
+                }
+            }
```

Comment:
```suggestion
                Button("Delete All", role: .destructive) {
                    model.deleteAll()
                }
```

Deployment target is 15 already 👏

Reply:
Ah amazing :) This was a copy-paste from History debug menu so let me update it there too!